plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -108,6 +122,14 @@ def _handle_image_manifest(self):
         self.add_child(self.v2_step_get_local_units)
         self.add_child(AddUnits(step_type=constants.UPLOAD_STEP_SAVE))
 
+    # *************************************************************************************************************
+    def _handle_manifest_list(self):
+        """
+        Handles the upload of a v2 s2 docker manifest list
+        """
+        self.add_child(AddManifestList(constants.UPLOAD_STEP_MANIFEST_LIST))
```

Comment:
what about making 2 steps in which you would extract data from json file, validate the extracted data, add the new unit if all validation passed.
1) ProcessManifestList - 
2) AddManifestList